plugins/catalog/src/components/CatalogTable/CatalogTable.tsx

```diff
@@ -40,19 +40,29 @@ import Edit from '@material-ui/icons/Edit';
 import OpenInNew from '@material-ui/icons/OpenInNew';
 import Star from '@material-ui/icons/Star';
 import StarBorder from '@material-ui/icons/StarBorder';
-import { capitalize } from 'lodash';
+import { capitalize, isFunction } from 'lodash';
 import React, { ReactNode, useMemo } from 'react';
 import { columnFactories } from './columns';
 import { CatalogTableRow } from './types';
 import pluralize from 'pluralize';
 
+/**
+ * Typed columns function to dynamically render columns based on entities and chosen kind.
+ *
+ * @public
+ */
+export type ColumnsFunc = (
```

Comment:
I wonder if it could make sense to pass an object similar to what `useEntityList()` returns, as your main issue here is that you don't have access to `<EntityListProvider />` 🤔

Reply:
Your suggestion to move up the `<EntityListProvider />` would be nice but since `<CatalogIndexPage />` is a top level page in `<App />` I'm not sure it'd be possible

Reply:
Yep let's keep it like this, it'd be tricky to move the context in a useful way. Could we switch up the name to be a bit more specific, e.g. `CatalogTableColumnsFunc`?